code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -131,6 +162,16 @@ public void setTokenCreators(List<TokenCreator> tokenCreators) {
         this.tokenCreators = tokenCreators;
     }
 
+    public void deletePreviousToken(long authenticatedAsAccountId, long tokenAccountId) {
+        List<AuthToken> authTokenList = authTokenDao.getTokenByAuthenticatedAsAccountId(authenticatedAsAccountId, tokenAccountId);
```

Comment:
The getting and deleting of tokens should be done in a single DB transaction. Could you move all this logic down into the DAO? In fact, in the function, instead of doing 1 select query and N delete queries, you should be able to do this in a single delete query with a where clause